devito/ir/support/space.py

```diff
@@ -121,25 +112,24 @@ class Interval(AbstractInterval):
 
     Create an Interval of size:
 
-        dim.size + abs(upper - lower)
+        (dim.extreme_max - dim.extreme_min + 1) + (upper - lower)
     """
 
     is_Defined = True
 
     def __init__(self, dim, lower, upper, stamp=0):
-        assert is_integer(lower)
-        assert is_integer(upper)
+        assert is_integer(lower) or isinstance(lower, Expr)
```

Comment:
:-)

Reply:
yep :) as promised. Try to understand what I did to make it work "the right way". It's not trivial